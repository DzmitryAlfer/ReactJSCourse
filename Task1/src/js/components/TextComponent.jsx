export const TextComponent = ({text}) => (
    <span>{text}</span>
);