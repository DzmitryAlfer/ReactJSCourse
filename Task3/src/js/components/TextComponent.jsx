import React from 'react';

export const TextComponent = ({text}) => (<span>{text}</span>);