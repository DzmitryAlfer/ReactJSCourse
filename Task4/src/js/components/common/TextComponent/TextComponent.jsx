import React from 'react';
import PropTypes from 'prop-types';

export const TextComponent = ({text, className}) => (<span className={className}>{text}</span>);

TextComponent.defaultProps = {
    text: "",
    className: ""
};

TextComponent.propTypes = {
    text: PropTypes.string.isRequired,
    className: PropTypes.string.isRequired
};