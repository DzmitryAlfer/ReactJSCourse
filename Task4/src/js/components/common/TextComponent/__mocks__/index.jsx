import React from 'react';

export const TextComponent = ({text, className}) => (<span>TextComponent MOCK</span>);