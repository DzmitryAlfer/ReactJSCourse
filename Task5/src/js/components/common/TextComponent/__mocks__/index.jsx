import React from 'react';

export const TextComponent = () => (<span>TextComponent MOCK</span>);