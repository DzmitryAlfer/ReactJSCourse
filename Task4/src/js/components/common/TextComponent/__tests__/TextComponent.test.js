import React from 'react';
import Renderer from 'react-test-renderer';
import {TextComponent} from "../TextComponent";

describe('TextComponent', () => {
    test('Component render', () => {
        const tree = Renderer.create(
            <TextComponent text='some-text' className='component-class'/>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});