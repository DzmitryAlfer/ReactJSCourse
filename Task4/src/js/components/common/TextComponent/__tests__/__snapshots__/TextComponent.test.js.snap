// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent Component render 1`] = `
<span
  className="component-class"
>
  some-text
</span>
`;
